java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java

```diff
@@ -277,6 +281,14 @@ private void writeFromVectorIntoJson(Field field, FieldVector vector) throws IOE
               vectorBufferTmp.setLong(0, 0);
               writeValueToGenerator(bufferType, vectorBufferTmp, null, vector, i);
             }
+          } else if (bufferType.equals(SIZE)
+              && vector.getValueCount() == 0
+              && vector.getMinorType() == MinorType.LISTVIEW) {
+            // Empty vectors may not have allocated a sizes buffer
+            try (ArrowBuf vectorBufferTmp = vector.getAllocator().buffer(4)) {
+              vectorBufferTmp.setInt(0, 0);
+              writeValueToGenerator(bufferType, vectorBufferTmp, null, vector, i);
+            }
```

Comment:
It should be OK to have an empty sizes buffer? Why are we setting a single value?

Reply:
Need to test this.

Reply:
Gathering my thoughts, I think the reason for doing this was basically to accommodate what was done to offset. There the current logic sets the a buffer with a single value. I think I naively followed the same approach. You have a point here, let me further test this.

Reply:
I think it should not apply for list view. We shouldn't blindly apply the same hacks since that may cover up bugs.